### ISSUE TITLE
Bond 6.8.x doesn't compile due to OSX version requirement in Differ

### ISSUE THREAD

**Issue by npvisual** [DeclarativeHub/Bond#526]
# Problem Statement
While I was trying to import dependencies, via Carthage, in one of my projects that involves Bond, I noticed the following error : 

```
/Users/.../Documents/Development/Swift/FrameWork101/Carthage/Checkouts/Bond/Sources/Bond/Collections/ObservableArray.swift:26:8: error: module file's minimum deployment target is OS X v10.11: /Users/.../Library/Caches/org.carthage.CarthageKit/DerivedData/9.3_9E145/Bond/6.8.0/Build/Intermediates.noindex/ArchiveIntermediates/Bond-macOS/BuildProductsPath/Release/Differ.framework/Modules/Differ.swiftmodule/x86_64.swiftmodule
import Differ
       ^

** ARCHIVE FAILED **


The following build commands failed:
        CompileSwift normal x86_64
        CompileSwiftSources normal x86_64 com.apple.xcode.tools.swift.compiler
(2 failures)
```

# Versions

MacOS X : 10.13.4 (17E202)
Xcode : Version 9.3 (9E145)
Carthage : 0.29.0

## Cartfile

```
github "DeclarativeHub/ReactiveKit"
github "DeclarativeHub/Bond"
github "DeclarativeHub/Layoutless"

....
*** Checking out Bond at "6.8.1"
*** Checking out Layoutless at "0.1.5"
*** Checking out Differ at "1.2.2"
*** Checking out ReactiveKit at "v3.9.4"
```

# How to reproduce ?

Simply use the attached Cartfile in a new project and run the following at the command prompt : 
```
$ carthage update
```

You can also reproduce the problem easily by importing the project (or a fork) and trying to build the `Bond-macOS` target.

# How to fix ?

Simply changing to `MACOSX_DEPLOYMENT_TARGET = 10.11` in the project settings solves the problem.

**Comment by daviskoh** [comment]
im seeing something similar when using Cocoapods as well

**Comment by npvisual** [comment]
FYI. Running the following will bypass the issue in the macOS build until it's resolved : 
```
carthage bootstrap --platform ios
```

**Comment by srdanrasic** [comment]
Thanks @npvisual! This should now be fixed in 6.8.2. 

Using `platform` argument is still recommended though. There is no need to build architectures you are not deploying to.
